src/server/appserver.cpp

```diff
@@ -3092,20 +3082,20 @@ void AppServer::handleCrashRecovery(bool &shouldQuit) {
         LOG_INFO(_logger, "Last session requested to not restart the server.");
         shouldQuit = _crashRecovered;
         if (!KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastServerSelfRestartDate, 0, found) || !found) {
-            LOG_ERROR(_logger, "Error in ParmsDb::updateAppState");
+            LOG_WARN(_logger, "Error in ParmsDb::updateAppState");
         }
         return;
     }
 
     std::string timestampStr;
     if (_crashRecovered) {
-        LOG_WARN(_logger, "Server auto restart after a crash.");
+        LOG_INFO(_logger, "Server auto restart after a crash.");
 
         if (serverCrashedRecently()) {
-            LOG_FATAL(_logger, "Server crashed twice in a short time, exiting");
+            LOG_INFO(_logger, "Server crashed twice in a short time, exiting");
```

Comment:
This is a serious problem. Level INFO seems a bit low.

Reply:
The problem is the crash that occurred at the previous launch of the app.
Here we just exit the app because it has been restarted twice in a short time.